cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -305,6 +321,8 @@ private void fixAnomaly(Anomaly anomaly) throws Exception {
           _anomalyDetectorState.onAnomalyHandle(anomaly, startedSuccessfully ? AnomalyState.Status.FIX_STARTED
                                                                              : AnomalyState.Status.FIX_FAILED_TO_START);
           LOG.info("Self-healing {}.", startedSuccessfully ? "started successfully" : "failed to start");
+          boolean isSuccessful = startedSuccessfully;
```

Comment:
redundant local variable `isSuccessful`

Reply:
needed because `_loggerExecutor.submit(() -> maybePersistSelfHealingInfo(anomaly, isSuccessful));` needed argument to be final or effectively final

Reply:
`isSuccessful` is a primitive type, IIUC it is copy-by-value in function call.

Reply:
IDE doesn't allow using `startedSuccessfully` directly.

Reply:
talked offline, it is required to pass  in an effectively final variable.